def faulty_function():
    print("Executing faulty_function...")
    return undefined_variable  # This will raise a NameError